internal/services/controller/handlers/transformers/deletedfinalstateunknown/transformer.go

```diff
@@ -0,0 +1,18 @@
+package deletedfinalstateunknown
+
+import (
+	"k8s.io/client-go/tools/cache"
+
+	"castai-agent/internal/castai"
+	"castai-agent/internal/services/controller/handlers/transformers"
+)
+
+func New() transformers.Transformer {
+	return func(e castai.EventType, obj interface{}) (castai.EventType, interface{}) {
+		if obj, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+			return castai.EventDelete, obj.Obj
+		}
+
+		return e, obj
+	}
+}
```

Comment:
maybe would be simpler to just declare a function of Transformer compatible type.

```suggestion
func DeleteFinalStateUnknown(e castai.EventType, obj interface{}) (castai.EventType, interface{}) {
		if obj, ok := obj.(cache.DeletedFinalStateUnknown); ok {
			return castai.EventDelete, obj.Obj
		}

		return e, obj
	}
```

Reply:
Ah, yes.